src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -443,5 +458,182 @@ internal static bool ContainsPathComponent(string path, string component, bool i
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a path relative to a directory.
+        /// </summary>
+        public static string GetRelativePath(string directory, string fullPath)
+        {
+            string relativePath = string.Empty;
+
+            if (IsChildPath(directory, fullPath))
+            {
+                return GetRelativeChildPath(directory, fullPath);
+            }
+
+            var directoryPathParts = GetPathParts(directory);
+            var fullPathParts = GetPathParts(fullPath);
+
+            if (directoryPathParts.Length == 0 || fullPathParts.Length == 0)
+            {
+                return fullPath;
```

Comment:
I'm always exception happy. Developers should never be allowed to get away with things!